src/cript/nodes/primary_nodes/project.py

```diff
@@ -76,6 +80,37 @@ def __init__(
         self._json_attrs = replace(self._json_attrs, name=name, collections=collections, material=material)
         self.validate()
 
+    def validate(self):
+        # First validate like other nodes
+        super().validate()
+
+        # Check graph for orphraned nodes, that should be listed in project
+        # Project.materials should contain all material nodes
+        all_materials = self.find_children({"node": ["Material"]})
+        for material in all_materials:
+            if material not in self.materials:
+                raise CRIPTOrphranedMaterialError(material)
+
+        # Check graph for orphraned nodes, that should be listed in the experiments
+        all_experiments = self.find_children({"node": ["Experiment"]})
```

Comment:
small picky preference, but I would change this into something like

```python
all_experiments = self.find_children(node="Experiment")
```
I think thats just nicer and cleaner, and under the hood find_children puts the experiment within brackets and gets it prepared and searches for it

Reply:
Unfortunately, this doesn't work.

This is just a simplified use of `find_children` it can do much more.
For example with `{"node": ["Experiment"], "name": "my name"}` you can find an experiment with a name too.
this also works for nested things.

It might be able to do this with `kwargs` but I don't want to open this in this PR.
Maybe a separate PR can address this.

Reply:
yeah I agree, this is a separate PR thing, but just wanted to make a small suggestion for future. 

It might be a good idea to not have the arguments be a dictionary and instead be explicit arguments and as simple as possible like:

```python
all_experiments = self.find_children(node="Experiment", name="my_name", )
```

I just really dislike `**kwargs` because its less explicit and if I am the user then I then have to go to documentation to see what is allowed and see how I can get clever about it to make it do what I need and if I missed anything. dictionary arguments are very open ended in my opinion, whereas explicit arguments you can't mess it up, and everything is right there and obvious . 

Unless there are too many possibilities where the function would have to have a million arguments to capture all possible arguments and situations that the function can be used in, and then a `**kwargs`  argument makes more sense.

I'm getting too far into it, but just wanted to put some food for thought and some ideas for later and we can come back to this later if we need to as well